### ISSUE TITLE
CircleCI does not fail on electron-builder errors

### ISSUE THREAD

**Issue by astralbodies** [Automattic/wp-desktop#468]
Looks like the artifacts stage is missing the Debian linux binary:

https://circleci.com/gh/Automattic/wp-desktop/19904#artifacts/containers/0

<img width="697" alt="2018-07-03_15-19-18" src="https://user-images.githubusercontent.com/373903/42242758-9edc7dc4-7ed4-11e8-9a4a-e3d37cd91806.png">

**Comment by adlk** [comment]
This is actually an issue with the CircleCI setup as the linux build failed but the job was not exited. 
```
Error: Exit code: 2. Command failed: /root/.cache/electron-builder/fpm/fpm-1.9.3-2.3.1-linux-x86_64/fpm -s dir -t deb --architecture amd64 --name WordPressDesktop --force --after-install /tmp/temp-dir-ulVQdY/0-after-install --after-remove /tmp/temp-dir-ulVQdY/1-after-remove --description WordPress.com Desktop client
 WordPress.com Desktop Client --version 3.4.0-beta1 --package /wp-desktop/release/WordPressDesktop_3.4.0-beta1_amd64.deb --maintainer Automattic Inc. <wp-desktop@automattic.com> --url https://desktop.wordpress.com/ --vendor Automattic Inc. <wp-desktop@automattic.com> --deb-compression xz --depends gconf2 --depends gconf-service --depends libnotify4 --depends libxtst6 --depends libnss3 --license GPLv2 --iteration 19902 --name=wordpress.com /wp-desktop/release/linux-unpacked/=/opt/WordPress.com /wp-desktop/release/.icon-set/icon_16x16.png=/usr/share/icons/hicolor/16x16/apps/wpcom.png /wp-desktop/release/.icon-set/icon_24x24.png=/usr/share/icons/hicolor/24x24/apps/wpcom.png /wp-desktop/release/.icon-set/app-icon_32.png=/usr/share/icons/hicolor/32x32/apps/wpcom.png /wp-desktop/release/.icon-set/icon_48x48.png=/usr/share/icons/hicolor/48x48/apps/wpcom.png /wp-desktop/release/.icon-set/app-icon_64.png=/usr/share/icons/hicolor/64x64/apps/wpcom.png /wp-desktop/release/.icon-set/icon_96x96.png=/usr/share/icons/hicolor/96x96/apps/wpcom.png /wp-desktop/release/.icon-set/app-icon_128.png=/usr/share/icons/hicolor/128x128/apps/wpcom.png /wp-desktop/release/.icon-set/app-icon_256.png=/usr/share/icons/hicolor/256x256/apps/wpcom.png /wp-desktop/release/.icon-set/app-icon_512.png=/usr/share/icons/hicolor/512x512/apps/wpcom.png /wp-desktop/release/.icon-set/app-icon_1024.png=/usr/share/icons/hicolor/1024x1024/apps/wpcom.png /tmp/temp-dir-ulVQdY/5-WordPress.com.desktop=/usr/share/applications/wpcom.desktop
/root/.cache/electron-builder/fpm/fpm-1.9.3-2.3.1-linux-x86_64/lib/ruby/bin/ruby: eval: line 5: syntax error near unexpected token `newline'
```
More info: [CircleCI -> Build Windows & Linux](https://circleci.com/gh/Automattic/wp-desktop/19902)

**Comment by sirreal** [comment]
This was fixed by 5cd95d6e87e16f6cd12ee9e5050fbab411d9256c which reverted #466.

The issue appears to be related to `nvm` as part of bash startup across CI jobs, which may not be the best approach. It's unclear exactly why that caused an issue.

#470 is an alternative approach to #466. `nvm` logic can be similarly abstracted and shared across jobs in another PR.
